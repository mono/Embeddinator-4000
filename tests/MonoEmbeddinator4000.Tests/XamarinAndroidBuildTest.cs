using NUnit.Framework;
using System.IO;
using System.CodeDom.Compiler;
using System.Collections.Generic;

namespace Embeddinator.Tests
{
    [TestFixture]
    public class XamarinAndroidBuildTest : UniverseTest
    {
        string manifestPath;

        public override void SetUp()
        {
            base.SetUp();

            temp = Path.Combine(Path.GetTempPath(), "hello.dll");
            manifestPath = Path.GetTempFileName();

            tempFiles = new List<string> { temp, manifestPath };
        }

        void GenerateAssembly()
        {
            var parameters = new CompilerParameters
            {
                OutputAssembly = temp,
            };
            AssemblyGenerator.CreateFromResource("Hello", parameters);
        }

        [Test]
        public void AndroidManifest()
        {
            GenerateAssembly();
            XamarinAndroidBuild.GenerateAndroidManifest(new[] { universe.LoadFile(temp) }, manifestPath, true);
            Approvals.VerifyFile(manifestPath);
        }

        [Test]
        public void AndroidManifestWithDots()
        {
            temp = Path.Combine(Path.GetTempPath(), "hello.with.dots.dll");
            tempFiles.Add(temp);

            GenerateAssembly();
            XamarinAndroidBuild.GenerateAndroidManifest(new[] { universe.LoadFile(temp) }, manifestPath, true);
            Approvals.VerifyFile(manifestPath);
        }

        [Test]
        public void AndroidManifestWithUpperCase()
        {
            temp = Path.Combine(Path.GetTempPath(), "HELLO.dll");
            tempFiles.Add(temp);

            GenerateAssembly();
            XamarinAndroidBuild.GenerateAndroidManifest(new[] { universe.LoadFile(temp) }, manifestPath, true);
            Approvals.VerifyFile(manifestPath);
        }
    }
}
