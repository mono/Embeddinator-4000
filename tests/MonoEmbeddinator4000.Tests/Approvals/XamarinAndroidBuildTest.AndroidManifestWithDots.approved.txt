<?xml version="1.0" encoding="utf-8"?>
<manifest xmlns:android="http://schemas.android.com/apk/res/android" package="hello_with_dots">
    <uses-sdk android:minSdkVersion="9" android:targetSdkVersion="24" />
    <application>
        <meta-data android:name="mono.embeddinator.classname" android:value="hello_with_dots.Native_hello_with_dots" />
        <provider android:name="mono.embeddinator.AndroidRuntimeProvider" android:exported="false" android:initOrder="2147483647" android:authorities="${applicationId}.mono.embeddinator.AndroidRuntimeProvider.__mono_init__" />
    </application>
</manifest>